### PR TITLE
Uprobe support

### PR DESCRIPTION
**Pull request by drzaeus77**
This adds uprobes support in two commits.

### CODE REVIEW

**Comment by drzaeus77** [comment]
Please leave comments on the attach_uprobe calling convention. Docstring will be my first change.

**Comment by drzaeus77** [comment]
Looks like the buildbots don't support uprobes. Might need to upgrade the kernel.

File: tests/cc/test_uprobes.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import bcc
+import ctypes
+import os
+import re
+import struct
+import time
+import unittest
+
+## code from ctypes impl
```
Comment (4ast) [inline tests/cc/test_uprobes.py]: these lines are left over?

Comment (drzaeus77) [inline tests/cc/test_uprobes.py]: Thanks, removed.

**Comment by drzaeus77** [comment]
@4ast this code should be ready now

File: src/python/bcc/__init__.py [hunk 2]
```diff
@@ -670,6 +692,141 @@ def detach_kretprobe(event):
...
+        cls._libsearch_cache[name] = path
+        return path
+
+    @classmethod
+    def find_load_address(cls, path):
+        if path in cls._lib_load_address_cache:
+            return cls._lib_load_address_cache[path]
+
+        # "LOAD off    0x0000000000000000 vaddr 0x0000000000400000 paddr 0x..."
+        with os.popen("""/usr/bin/objdump -x %s | awk '$1 == "LOAD" && $3 ~ /^[0x]*$/ { print $5 }'""" % path) as f:
```
Comment (4ast) [inline src/python/bcc/__init__.py]: either full path to objump in both places or both without?

Comment (drzaeus77) [inline src/python/bcc/__init__.py]: Fixed in next patch.

**Comment by 4ast** [comment]
Nice!

**Pull request merged**